mods/player_api/api.lua

```diff
@@ -1,54 +1,85 @@
--- Minetest 0.4 mod: player
--- See README.txt for licensing and other information.
-
 player_api = {}
 
 -- Player animation blending
 -- Note: This is currently broken due to a bug in Irrlicht, leave at 0
 local animation_blend = 0
 
-player_api.registered_models = { }
+player_api.registered_models = {}
 
 -- Local for speed.
 local models = player_api.registered_models
 
+local function collisionbox_equals(collisionbox, other_collisionbox)
+	if collisionbox == other_collisionbox then
+		return true
+	end
+	for index = 1, 6 do
+		if collisionbox[index] ~= other_collisionbox[index] then
+			return false
+		end
+	end
+	return true
+end
+
 function player_api.register_model(name, def)
 	models[name] = def
+	def.visual_size = def.visual_size or {x = 1, y = 1}
+	def.collisionbox = def.collisionbox or {-0.3, 0.0, -0.3, 0.3, 1.7, 0.3}
+	def.stepheight = def.stepheight or 0.6
+	def.eye_height = def.eye_height or 1.47
+	for animation_name, animation in pairs(def.animations) do
+		animation.eye_height = animation.eye_height or def.eye_height
+		animation.collisionbox = animation.collisionbox or def.collisionbox
+		for _, other_animation in pairs(def.animations) do
+			if other_animation._equals then
+				if collisionbox_equals(animation.collisionbox, other_animation.collisionbox)
+						and animation.eye_height == other_animation.eye_height then
+					animation._equals = other_animation._equals
+					break
+				end
+			end
+		end
+		animation._equals = animation._equals or animation_name
+	end
 end
 
 -- Player stats and animations
-local player_model = {}
-local player_textures = {}
-local player_anim = {}
-local player_sneak = {}
+-- model, textures, animation
+local players = {}
 player_api.player_attached = {}
 
+local function get_player_data(player)
+	return assert(players[player:get_player_name()], "offline_player")
```

Comment:
It is human readable, as a warning is logged (code below). Using the error flow is required to avoid code duplication. If you don't like this, I can also throw an error table and do an ID comparison.

Reply:
¯\_(ツ)_/¯ well then. good enough I guess...